src/client/Client.js

```diff
@@ -440,8 +440,8 @@ class Client extends BaseClient {
    * @private
    */
   _validateOptions(options = this.options) {
-    if (typeof options.ws.intents !== 'undefined') {
-      options.ws.intents = Intents.resolve(options.ws.intents);
+    if (typeof options.intents !== 'undefined') {
+      options.ws.intents = Intents.resolve(options.intents);
     }
```

Comment:
```suggestion
    if (typeof options.intents === 'undefined') {
      throw new TypeError('CLIENT_INAVLID_OPTION', 'intents', 'a string, a number, an Intents object, or an Array');
    } else {
      options.ws.intents = Intents.resolve(options.intents);
    }
```

Reply:
i decided to make a new error `CLIENT_MISSING_INTENTS` and also prevent passing `0` intents